In [1]:
import numpy as np

Задание 1.Провести дисперсионный анализ для определения того,

есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов.

Даны значения роста в трех группах случайно выбранных спортсменов:

Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. soccer (sc)

Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. hockey (hk)

Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170. barbell (bb)

In [2]:
sc = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hk = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
bb = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [3]:
all_mean = (np.sum(sc)+np.sum(hk)+np.sum(bb)) / (np.size(sc)+ np.size(hk)+np.size(bb))
# сумма произведений квадратов отклонений средних по группам от общего среднего и наблюдений в группе
fact_dev = (np.mean(sc)-all_mean)**2*np.size(sc) + (np.mean(hk)-all_mean)**2*np.size(hk) + (np.mean(bb)-all_mean)**2*np.size(bb)
# факторная дисперсия
fact_var = fact_dev/2

In [4]:
# остаточная сумма квадратов отклонений
resid_dev = np.sum((sc - np.mean(sc))**2) + np.sum((hk - np.mean(hk))**2) + np.sum((bb - np.mean(bb))**2)
# Остаточная дисперсия
resid_var = resid_dev / ((np.size(sc)+ np.size(hk)+np.size(bb)) - 3)

In [5]:
all_dev = np.sum((sc-all_mean)**2) + np.sum((hk-all_mean)**2) + np.sum((bb-all_mean)**2)
all_dev

830.9642857142858

In [6]:
fact_dev + resid_dev

830.964285714286

In [7]:
f_h = fact_var / resid_var
f_h

5.500053450812598

In [8]:
df_1 = 2
df_2 = ((np.size(sc)+ np.size(hk)+np.size(bb)) - 3)
df_2

25

In [9]:
# while alpha = 0.05, df1 = 2, df2 = 25, f_cr = 3.3852
# while alpha = 0.01, df1 = 2, df2 = 25, f_cr = 5.568

При критической области 5% фактор вида спорта показывает различия среднего роста спортсмена (критерий 5.5 больше значения критической точки 3.3852),
а при 1% - эти различия входят в разброс предсказания математического ожидания генеральной совокупности. 
